internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -288,6 +281,70 @@ func createRun(opts *CreateOpts) error {
 				return err
 			}
 		}
+	case Dynamic:
+		sc, err := readConfigFile(opts)
+		if err != nil {
+			return fmt.Errorf("failed to process config file: %w", err)
+		}
+
+		missingFields := validateFields(sc)
+		if len(missingFields) > 0 {
+			return fmt.Errorf("missing required field(s) in the config file: %s", missingFields)
+		}
+
+		switch sc.Type {
+		case integrations.AWS:
+
+			missingDetails := validateDetails(sc.Details, AwsKeys)
+
+			if len(missingDetails) > 0 {
+				return fmt.Errorf("missing required field(s) in the config file details: %s", missingDetails)
+			}
+
+			req := preview_secret_service.NewCreateAwsDynamicSecretParamsWithContext(opts.Ctx)
+			req.OrganizationID = opts.Profile.OrganizationID
+			req.ProjectID = opts.Profile.ProjectID
+			req.AppName = opts.AppName
+			req.Body = &preview_models.SecretServiceCreateAwsDynamicSecretBody{
+				IntegrationName: sc.IntegrationName,
+				DefaultTTL:      sc.Details[AwsKeys[0]].(string),
+				AssumeRole: &preview_models.Secrets20231128AssumeRoleRequest{
+					RoleArn: sc.Details[AwsKeys[1]].(string),
+				},
```

Comment:
I wonder if we'd be able to unmarshal the serialized data from the file directly into the API structures like `preview_models.Secrets20231128AssumeRoleRequest`? Relying on the order of values in `AwsKeys` seems like it could become error prone.

Reply:
I think this is possible, but would take some massaging. Right now the secret config is
```
type SecretConfig struct {
	Version         string
	Type            integrations.IntegrationType
	IntegrationName string `yaml:"integration_name"`
	Details         map[string]any
}
```
with the idea of trying to keep the . There is an `UnmarshalBinary` function that can be used on proto messages, but it takes a byte array. When the file is first read, it's gathered as a []byte, but there are relevant fields (like `name`) not provided in the file, and fields that wouldn't serialize over (like secret provider `type`). Kinda tricky since we've tried to keep the structure here as generic as possible so that rotating and dynamic secrets are supported, plus all providers. But I def agree that the current hardcoded configuration is pretty brittle. Let me think about this a bit more.